mkosi/config.py

```diff
@@ -1765,6 +1773,36 @@ def backward_compat_stubs(self, namespace: argparse.Namespace) -> None:
             delattr(namespace, "cache")
             print("Warning: --cache is no longer supported")
 
+    def resolve_deps(self, args: MkosiArgs, presets: Sequence[MkosiConfig]) -> list[MkosiConfig]:
+        graph = {p.preset: p.depends for p in presets}
+
+        if args.presets:
+            for p in args.presets:
+                if p not in graph:
+                    die(f"No preset found with name {p}")
+
+            deps = set()
+            queue = [*args.presets]
+
+            while queue:
+                preset = queue.pop()
+                if preset in deps:
+                    continue
+
+                deps.add(preset)
+                queue += [*graph[preset]]
```

Comment:
Maybe a bit pedantic, but it's a stack, not a queue. `list.pop()` will pop the last element (on the right side) and this will add new elements on the right. 

Also, with a few less sigils this is
```suggestion
                queue.extend(graph[preset])
```
but generally this would probably be better with a `collections.deque` for `queue`, which could do a `popleft` above then.